src/PIL/ImageFont.py

```diff
@@ -259,9 +259,10 @@ def getsize(
 
         :return: (width, height)
         """
+        # vertical offset is added for historical reasons, see discussion in #4789
         size, offset = self.font.getsize(text, False, direction, features, language)
         return (
-            size[0] + stroke_width * 2 + offset[0],
+            size[0] + stroke_width * 2,
             size[1] + stroke_width * 2 + offset[1],
         )
```

Comment:
This PR fixes an issue introduced in #2576 (Pillow 4.2.0) where the x offset was sometimes added (subtracted) twice in the C function, see https://github.com/python-pillow/Pillow/issues/4789#issuecomment-659609574. This change adjusts the Python function to match this behaviour after fixing the C function.

The offset was first added in #784 to match pre-#45/#185 behaviour (and make `getsize` return the bottom right text coordinate, which used to match the size), but #2576 effectively reverted this change for the horizontal axis **when the first character has a negative X bearing in horizontal text** (otherwise the x offset is 0).

I think it is better to stick with the current behaviour (which returns the actual width but might be confusing if the first character has a negative bearing) and document it (in part 3 of #4724) rather than reverting back to the previous behaviour, as it has been this way for a long time now.

*I will update the comment to link to this explanation for future reference.*